### PR TITLE
Update Model Class Diagram in DG

### PR DESCRIPTION
**Pull request by CrownKira**
### Description

What's new 
- Updates model class diagram and its corresponding puml file
- Adds Field class diagram 
- Adds new sections about field interfaces

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### Checklist

- [ ] I have tested this code
- [x] I have updated the documentation

### CODE REVIEW

File: docs/diagrams/ModelClassDiagram.puml [hunk 1]
```diff
@@ -38,17 +39,20 @@ ModelManager -left-> "1" AddressBook
 ModelManager -right-> "1" UserPrefs
 UserPrefs .up.|> ReadOnlyUserPrefs
 
-AddressBook *--> "1" UniquePersonList
-UniquePersonList --> "~* all" Person
```
Comment (benedictchuajj) [inline docs/diagrams/ModelClassDiagram.puml:42]: Should this be deleted? I'm not sure if we changed the relationship for UniquePersonList --> Person, so do let me know if it has indeed changed or if it's better to leave this out.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -154,6 +148,52 @@ The `Storage` component,
...
+
+We provide options for developers to easily customise the constraints on the user input such as whether the input is required or whether it is editable. These field options are encapsulated within the `Field` interface, which further branches into more concrete interfaces which can be implemented by `Person` attributes.
+
+Field options largely dictate how the parsers respond to user's inputs.
+
+Option | Description
+--- | --- 
+IS_BLANK_VALUE_ALLOWED | If set to `true`, the field is allowed to be blank (for string fields such as phone, name, etc)
+IS_NULL_VALUE_ALLOWED | If set to `true`, the field is allowed to be null (for int/Date fields such as LastMet, etc)
+DEFAULT_VALUE | The default value for the field. Set when user does not pass in the prefix on `Person` creation.
```
Comment (benedictchuajj) [inline docs/DeveloperGuide.md:161]:
```suggestion
DEFAULT_VALUE | The default value for the field. Set when user does not pass in any values for the respective prefix on `Person` creation.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -154,6 +148,52 @@ The `Storage` component,
...
+DEFAULT_VALUE | `""`
+IS_EDITABLE | `true`
+
+#### RequiredField
+
+Option | Default
+--- | --- 
+IS_BLANK_VALUE_ALLOWED | `false`
+IS_NULL_VALUE_ALLOWED | `false`
+DEFAULT_VALUE | `""`
```
Comment (benedictchuajj) [inline docs/DeveloperGuide.md:194]: In addition to the actual default value, should we state that Default Value isn't applicable (based on the description above) for RequiredFields?
```suggestion
DEFAULT_VALUE | `""` (Not Applicable)
```

File: docs/diagrams/FieldClassDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+skinparam arrowColor MODEL_COLOR
+skinparam classBackgroundColor MODEL_COLOR
+
+Interface Field <<Interface>>
+Interface OptionalStringBasedField <<Interface>>
+Interface OptionalNonStringBasedField <<Interface>>
+Interface RequiredField <<Interface>>
+
+OptionalStringBasedField -up-|> Field
+OptionalNonStringBasedField -up-|> Field
```
Comment (benedictchuajj) [inline docs/diagrams/FieldClassDiagram.puml:13]: Perhaps this should be: 
```suggestion
OptionalPersonStringField -up-|> Field
OptionalPersonNonStringField -up-|> Field
```


Feel free to ignore if we are changing the interface names to fit the DG. If you were to change the naming here, do remember to change the naming in DG.md too!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (CrownKira) [reply docs/DeveloperGuide.md:161]: Thanks for correcting my poor English. Feel free to directly make a commit to this branch.

File: docs/diagrams/ModelClassDiagram.puml (hunk 1 above)
Comment (CrownKira) [reply docs/diagrams/ModelClassDiagram.puml:42]: No we didnt?

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (CrownKira) [reply docs/DeveloperGuide.md:161]: Edit: I actually really meant does not pass in prefix. instead of value for prefix.

**Pull request merged**